spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/RemoveCarryoverIterator.java

```diff
@@ -151,13 +153,21 @@ private int[] generateIndicesToIdentifySameRow(int columnSize) {
     return indices;
   }
 
-  private boolean isSameRecord(Row currentRow, Row nextRow) {
-    for (int idx : indicesToIdentifySameRow) {
+  protected boolean isSameRecord(Row currentRow, Row nextRow) {
```

Comment:
Can we move this to base class?

Reply:
The class `ComputeUpdateIterator` doesn't need it. Can we do that until all subclasses of `ChangelogIterator` need it?

Reply:
Code-wise, I am thinking that it is a bit harder to read to have RemoveNetCarryoverIterator to extend RemoveCarryoverIterator.  So was thinking we can move have an extra base class, then its easier to see what is the different methods and what is same, not sure what you think.  Now its a bit trickier to see that.

Reply:
Move it to ChangelogIterator, and let RemoveNetCarryoverIterator inherit ChangelogIterator directly. It should be easier to read.